aten/src/ATen/native/transformers/attention.cpp

```diff
@@ -628,8 +628,10 @@ std::tuple<Tensor, Tensor> _scaled_dot_product_attention_math(
     // Naive, composite implementation defined here.
 
     // Scale q, k before matmul for stability see https://tinyurl.com/sudb9s96 for math
-    const auto scaling_factor = sdp::calculate_scale(query_, scale).sqrt();
-    const auto query = query_ * scaling_factor;
+    bool is_negative_scaling = scale.has_value() && scale.value() < 0.0;
+    const auto scaling_factor = sdp::calculate_scale(query_, is_negative_scaling ? std::abs(scale.value()) : scale).sqrt();
+
+    const auto query = query_ * (is_negative_scaling ? c10::SymFloat(0.0) - scaling_factor: scaling_factor);
```

Comment:
Why doing all this here rather than teach `sdp::calculate_scale` how to handle negative scales?

Reply:
Now, that being said, do we want the CPU path to deviate from the CUDA path?

Reply:
So we need to make sure the ```_scaled_dot_product_attention_math``` will get the same value with the cuda fused ```_scaled_dot_product_attention``` path, is it right?

Reply:
Yeah, I believe so (modulo precision errors, of course)